sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/DotNetSensorTest.java

```diff
@@ -117,25 +118,35 @@ public void noProtobufFilesShouldNotFail() {
 
     sensor.execute(tester);
 
-    assertThat(logTester.logs(LoggerLevel.WARN)).containsOnly("No protobuf reports found - no metrics and highlighting will be imported.");
+    assertThat(logTester.logs(LoggerLevel.WARN)).containsExactly(
+      "No protobuf reports found. The " + LANG_NAME + " files will not have highlighting and metrics.",
+      "Your project contains " + LANG_NAME + " files which cannot be analyzed with the scanner you are using." +
+        " To analyze C# or VB.NET files, you must use the Scanner for MSBuild 4.x, read more on https://redirect.sonarsource.com/doc/install-configure-scanner-msbuild.html ." +
+        " For any questions you may have, open a topic on https://community.sonarsource.com .");
     verify(reportPathCollector).protobufDirs();
     verifyZeroInteractions(protobufDataImporter);
     ImmutableMap<String, List<RuleKey>> expectedMap = ImmutableMap.of(
-      "sonaranalyzer-langKey", ImmutableList.of(RuleKey.of(REPO_KEY, "S1186"), RuleKey.of(REPO_KEY, "[parameters_key]")),
+      "sonaranalyzer-" + LANG_KEY, ImmutableList.of(RuleKey.of(REPO_KEY, "S1186"), RuleKey.of(REPO_KEY, "[parameters_key]")),
       "foo", ImmutableList.of(RuleKey.of("roslyn.foo", "custom-roslyn")));
     verify(roslynDataImporter).importRoslynReports(eq(Collections.singletonList(new RoslynReport(null, workDir.getRoot()))), eq(tester), eq(expectedMap),
       any(RealPathProvider.class));
   }
 
   @Test
-  public void noRoslynReportShouldFail() {
+  public void noRoslynReportShouldNotFail() {
     addFileToFs();
-    assertThatExceptionOfType(IllegalStateException.class).isThrownBy(() -> sensor.execute(tester))
-      .withMessage("No Roslyn issue reports were found.");
+
```

Comment:
_Nitpick:_ Can this empty line be removed? Both lines seems to be part of orchestration before acting.